### ISSUE TITLE
is this a screensaver? where is the file after building? lol

### ISSUE THREAD

**Issue by i3roly** [sandydoo/flux#25]
hi,

so apparently i've built the screensaver (this is a screensaver, right)?

i ran a cargo build in both the flux-desktop, flux and web directories (separately) and i see nuffin.

i also did the recommended pnpm build etc in the web dir, still nothing.

i was hoping to have something i could use as a screensaver for mac.

**Comment by i3roly** [comment]
my bad i found it, it's an executable in the target/debug directly

but isn't there a way to package this into a nice .saver form so that we can use it on mac?

**Comment by sandydoo** [comment]
> so apparently i've built the screensaver (this is a screensaver, right)?

This is the source code for the library and the web version. The primary goal is to facilitate learning and showcase what you can build with rust.

> but isn't there a way to package this into a nice .saver form so that we can use it on mac?

I've decided to make the native screensavers private because nobody is interested in helping maintain them. What everyone wants is a pre-built binary and dedicated support when something doesn't work. This is not sustainable for me.

I sell the Windows version to offset the development, hardware, and support costs. I may offer a macOS version once the feature set exceeds what is already available in Drift.

**Comment by i3roly** [comment]
i run 10.14 though, which is why i am asking for a mac version.

is it even possible to create a macos screensaver with a binary? it doesn’t seem straightforward but i could be wrong.

Thanks,
Gagan

**Comment by sandydoo** [comment]
You would need to write an FFI library, compile it, link it in Xcode, initialise the appropriate graphics library, and then wire up ScreensaverView.

**Comment by i3roly** [comment]
yeah I thought it wasn't trivial.  a little too much work for me lol. Thanks for your efforts. Do let me know if you create a mac version, but I doubt it's worth it since you'd be catering to pre 10.15 machines.‎My next box is gonna be a win{doze,blowz,etc} one anyways, given what that special student cook has done using those piece of shit arm processors. Looking forward to coming back with each day. Only thing I don't like is the line endings discrepancy. Here's to hoping microsoft actually leverages its posix compliance and gives us a true posix compliance shell, instead of wsl. That'd be so dope                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone on the Rogers network.                                                                                                                                                                                                                From: sanderSent: Monday, 10 June 2024 8:49 AMTo: sandydoo/fluxReply To: sandydoo/fluxCc: gagan sidhu; AuthorSubject: Re: [sandydoo/flux] is this a screensaver? where is the file after building? lol (Issue #25)
You would need to write an FFI library, compile it, link it in Xcode, initialise the appropriate graphics library, and then wire up ScreensaverView.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/sandydoo/flux/issues/25#issuecomment-2158573093",
"url": "https://github.com/sandydoo/flux/issues/25#issuecomment-2158573093",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
